visualization/CMakeLists.txt

```diff
@@ -10,6 +10,13 @@ find_package(catkin REQUIRED COMPONENTS
 	rviz
 )
 
+# rviz transitively includes OGRE headers which break with `-Wall -Werror`
+# so isolate this include dir and add it as SYSTEM include where needed.
+find_path(OGRE_INCLUDE_DIR OgreRoot.h
```

Comment:
Note, that rviz currently includes two OGRE folders: `/usr/include/OGRE/Overlay` and `/usr/include/OGRE`.
I think you should handle both in the same fashion. At first sight, I considered this approach elegant.
However, it is also brittle, because any change of OGRE include folders in rviz, will probably break here.